File: solutions/c/grains/1/grains.c
#include "grains.h"
#include <math.h>

uint64_t square(uint8_t n) {
    return pow(2, n - 1);    
}

uint64_t total() {
    int total = 0;
    for (int i = 1; i < 65; i ++) {
        total += square(i);
    }
    return total;
}

File: solutions/java/grains/1/src/main/java/Grains.java
import java.math.BigInteger;

class Grains {

    BigInteger grainsOnSquare(final int square) {
        if (square < 1 || square > 64) {
            throw new IllegalArgumentException("square must be between 1 and 64");
        }
        return BigInteger.valueOf(2).pow(square - 1);
    }

    BigInteger grainsOnBoard() {
        BigInteger sum = BigInteger.valueOf((long)(0));
        for (int i = 1; i<=64; i++) {
            sum = sum.add(grainsOnSquare(i));
        }
        return sum;
    }

}
